docs/content/guides/rows/rows-sorting.md

```diff
@@ -0,0 +1,3432 @@
+        {
+          title: 'Model',
+          type: 'text',
+          data: 'model',
+        },
+        {
+          title: 'Price',
+          type: 'numeric',
+          data: 'price',
+          numericFormat: {
+            pattern: '$ 0,0.00',
+            culture: 'en-US'
+          },
+          className: 'htLeft',
+        },
+        {
+          title: 'Date',
+          type: 'date',
+          data: 'sellDate',
+          className: 'htRight',
+        },
+        {
+          title: 'Time',
+          type: 'time',
+          data: 'sellTime',
+          className: 'htRight',
+        },
+        {
+          title: 'In stock',
+          type: 'checkbox',
+          data: 'inStock',
+          className: 'htCenter',
+        },
+      ]}
+      columnSorting={{
+        initialConfig: {
+          column: 1,
+          sortOrder: 'desc',
+        },
+      }}
+      className="custom-sort-icon-2"
+      licenseKey="non-commercial-and-evaluation"
+    />
+  );
+};
+
+/* start:skip-in-preview */
+ReactDOM.render(<HandsontableComponent />, document.getElementById('example7'));
+/* end:skip-in-preview */
+```
+
+```css
+/* the icon for ascending order */
+.custom-sort-icon-2 span.colHeader.columnSorting.ascending::before {
+  content: '△';
+  background-image: none !important;
+}
+
+/* the icon for descending order */
+.custom-sort-icon-2 span.colHeader.columnSorting.descending::before {
+  content: '▽';
+  background-image: none !important;
+}
+```
+
+:::
+
+:::
+
+## Add a custom comparator
+
+A comparator is a function that determines the sort order, based on specified criteria.
+
+Adding a custom comparator lets you go beyond Handsontable's built-in sorting features. You can:
+- Apply a custom sort order. For example, instead of sorting data alphabetically or numerically, you can sort it by
+  length or by the occurrence of a specific character.
+- Handle exceptions. For example, in a list of employees, you can exclude workers with a specific job title from sorting.
+- Implement a custom sorting logic based on your own criteria.
+
+To add a custom comparator, use the [`compareFunctionFactory`](@/api/options.md#columnsorting) option.
+
+::: only-for javascript
+
+```js
+const configurationOptions = {
+  columnSorting: {
+    compareFunctionFactory: function(sortOrder, columnMeta) {
+      // implement your own comparator
+      return function(value, nextValue) {
+        if (`value` is less than `nextValue` by some ordering criterion) {
+          return -1;
+        }
+        if (`value` is greater than `nextValue` by the same ordering criterion) {
+          return 1;
+        }
+        return 0;
+      }
+    },
+  },
+};
+```
+
+:::
+
+::: only-for react
+
+```jsx
+<HotTable
+  columnSorting={{
+    compareFunctionFactory: function(sortOrder, columnMeta) {
+      // implement your own comparator
+      return function(value, nextValue) {
+        if (`value` is less than `nextValue` by some ordering criterion) {
```

Comment:
I don't think that it's a good idea to keep pseudo/meta code in code snippets. It's not copy-pastable, the color syntax - which is broken, reinforces the idea that something is wrong with that snippet.